lib/srv/regular/sshserver.go

```diff
@@ -897,13 +928,18 @@ func (s *Server) handleSessionRequests(sconn *ssh.ServerConn, identityContext sr
 				req.Reply(true, nil)
 			}
 		case result := <-ctx.ExecResultCh:
-			ctx.Debugf("[SSH] ctx.result = %v", result)
-			// this means that exec process has finished and delivered the execution result,
-			// we send it back and close the session
+			ctx.Debugf("Exec request (%q) complete: %v", result.Command, result.Code)
+
+			// The exec process has finished and delivered the execution result, send
+			// the result back to the client, and close the session and channel.
 			_, err := ch.SendRequest("exit-status", false, ssh.Marshal(struct{ C uint32 }{C: uint32(result.Code)}))
 			if err != nil {
-				ctx.Infof("[SSH] %v failed to send exit status: %v", result.Command, err)
+				ctx.Infof("Failed to send exit status for %v: %v", result.Command, err)
 			}
+
+			return
+		case <-closeContext.Done():
+			log.Debugf("Closing session due to context closing.")
```

Comment:
let's make it more informative, how about `Closing session due to missed heartbeat` AFAIK this context is closed when heartbeat fails